src/test/CMakeLists.txt

```diff
@@ -244,6 +244,25 @@ add_dependencies(hyriseTest hyriseTestPlugin hyriseTestNonInstantiablePlugin)
 target_link_libraries(hyriseTest hyrise ${LIBRARIES})
 target_link_libraries_system(hyriseTest nlohmann_json::nlohmann_json)
 
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+target_compile_options(hyriseTest PUBLIC -Xclang -fno-pch-timestamp)
+endif()
+if(NOT ${CMAKE_VERSION} VERSION_LESS "3.16.0")
+    target_precompile_headers(hyriseTest PRIVATE
+        <gtest/gtest.h>
+        [["all_parameter_variant.hpp"]]
+        [["storage/create_iterable_from_segment.hpp"]]
+        [["storage/table.hpp"]]
+        [["types.hpp"]]
+    )
+
+    # Exclude files that use gmock as these are incompatible with unity builds and pre-compiled headers
+    set_source_files_properties(
+        server/server_session_test.cpp
```

Comment:
Please merge a recent master into this branch. This file is not part of the server implementation anymore. Also, AFAIK, we got more mock classes in our code base.